core/client/src/main/java/alluxio/client/file/FileOutStream.java

```diff
@@ -99,22 +98,21 @@ public FileOutStream(AlluxioURI path, OutStreamOptions options, FileSystemContex
     mCanceled = false;
     mShouldCacheCurrentBlock = mAlluxioStorageType.isStore();
     mBytesWritten = 0;
-    try {
-      if (!mUnderStorageType.isSyncPersist()) {
-        mUfsPath = null;
-        mUnderStorageOutputStream = null;
-        mFileSystemWorkerClient = null;
-      } else {
-        mUfsPath = options.getUfsPath();
-        mFileSystemWorkerClient = mCloser.register(mContext.createFileSystemWorkerClient());
-        mUnderStorageOutputStream = mCloser.register(UnderFileSystemFileOutStream.create(mContext,
-            mFileSystemWorkerClient.getWorkerDataServerAddress(), mUfsPath, options.getOwner(),
-            options.getGroup(), options.getMode()));
+
+    if (!mUnderStorageType.isSyncPersist()) {
+      mUnderStorageOutputStream = null;
+    } else { // Write is through to the under storage, create mUnderStorageOutputStream
+      try {
+        WorkerNetAddress worker = // not storing data to Alluxio, so block size is 0
+            options.getLocationPolicy().getWorkerForNextBlock(mBlockStore.getWorkerInfoList(), 0);
+        InetSocketAddress location = new InetSocketAddress(worker.getHost(), worker.getDataPort());
+        mUnderStorageOutputStream =
+            mCloser.register(UnderFileSystemFileOutStream.create(mContext, location, mOptions));
+      } catch (AlluxioException | IOException e) {
+        mCloser.close();
```

Comment:
Updated, thanks.
Should this be done everywhere we call `Closer#close`, if so could you open a ticket for this?